legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -167,15 +167,28 @@ def search_businesses():
     try:
         json_input = request.get_json()
         identifiers = json_input.get('identifiers', None)
-        if not identifiers or not isinstance(identifiers, list):
-            return {'message': "Expected a list of 1 or more for '/identifiers'"}, HTTPStatus.BAD_REQUEST
+        search_filter_name = json_input.get('name', None)
+        search_filter_type = json_input.get('type', None)
 
-        # base business query
-        bus_query = db.session.query(Business).filter(Business._identifier.in_(identifiers))  # noqa: E501; pylint: disable=protected-access
+        filters = []
 
-        # base filings query (for draft incorporation/registration filings -- treated as 'draft' business in auth-web)
-        draft_query = db.session.query(Filing).filter(
-            and_(Filing.temp_reg.in_(identifiers), Filing.business_id.is_(None)))
+        # Add filters if any
+        if identifiers and isinstance(identifiers, list):
+            filters.append(Business._identifier.in_(identifiers))
+
+        if search_filter_name:
+            filters.append(Business.legal_name.ilike(f'%{search_filter_name}%'))
```

Comment:
You'll need to do something else for `get_alternative_names` ? as this is what is shown not the `legal_name` for SP/GP